src/Microsoft.OpenApi.Readers/ParseNodes/ParseNode.cs

```diff
@@ -31,8 +31,9 @@ public MapNode CheckMapNode(string nodeName)
             var mapNode = this as MapNode;
             if (mapNode == null)
             {
-                Diagnostic.Errors.Add(
-                    new OpenApiError("", $"{nodeName} must be a map/object at " + Context.GetLocation()));
+                throw new OpenApiException($"{nodeName} must be a map/object");
```

Comment:
why change to throw exception?

Reply:
Because it caused a cascade of errors as following code attempted to handle the null node.